flocker/node/_deploy.py

```diff
@@ -642,19 +675,17 @@ def applications_from_units(units):
                     running=(unit.activation_state == u"active"),
                 ))
 
-            return NodeState(
+            return [NodeState(
                 hostname=self.hostname,
                 applications=applications,
                 used_ports=self.network.enumerate_used_ports(),
                 manifestations=None,
                 paths=None,
-            )
+            )]
         d.addCallback(applications_from_units)
         return d
 
-    def calculate_necessary_state_changes(self, local_state,
-                                          desired_configuration,
-                                          current_cluster_state):
+    def calculate_changes(self, desired_configuration, current_cluster_state):
```

Comment:
The docstring for this method still talks about volumes a little bit more than it should.